src/transformers/models/paligemma/modeling_paligemma.py

```diff
@@ -301,14 +301,15 @@ def _merge_input_ids_with_image_features(
         pad_mask = input_ids == self.pad_token_id
 
         # expand masks to match embedding dimension
-        text_mask_expanded = text_mask.unsqueeze(-1).expand(-1, -1, embed_dim)
-        pad_mask_expanded = pad_mask.unsqueeze(-1).expand(-1, -1, embed_dim)
+        text_mask_expanded = text_mask.unsqueeze(-1).expand(-1, -1, embed_dim).to(inputs_embeds.device)
+        pad_mask_expanded = pad_mask.unsqueeze(-1).expand(-1, -1, embed_dim).to(inputs_embeds.device)
```

Comment:
Ok this part I am not sure, the device that accelerate chooses for the input ids is usually automatically deduced? 
cc @SunMarc or @muellerzr ?

Reply:
Yeah, when `input_ids` is called in the forward of `PaliGemmaForConditionalGeneration`, it will be moved the [main device](https://github.com/huggingface/accelerate/blob/0553483638fcde878158acef9cfd6dbb62757515/src/accelerate/big_modeling.py#L391) thanks to a pre_forward hook (e.g. cuda:0) . However, since `inputs_embeds` is not necessarily on the same device as `input_ids` (e.g. cuda:1). Hence we can't really make sure that they are both on the same device.

Reply:
Right, that's what happened when I tried on a multi-gpu setting with `device_map` to `auto`, they were on different devices. And this hook can't apply to `inputs_embeds` because it is created later, is that correct?

Reply:
Yeah it's right !